src/internal/cluster/zarf.go

```diff
@@ -99,21 +102,111 @@ func (c *Cluster) StripZarfLabelsAndSecretsFromNamespaces() {
 	spinner.Success()
 }
 
+// PackageSecretNeedsWait checks if the the package status or any of the component statuses are being mutated by a webhook and need to be waited on.
+func (c *Cluster) PackageSecretNeedsWait(secretName string) (bool, int, error) {
```

Comment:
This is not a standard function signature - most other functions take the package name, not a pre-constructed secret name.

Reply:
Another option would be to pass in the secret object since we already have it on the call of this function.